frontend/src/pages/ActivityReport/Pages/goalsObjectives.js

```diff
@@ -117,8 +117,13 @@ const GoalsObjectives = ({
     if (index !== -1) {
       copyOfSelectedGoals.splice(index, 1);
     }
-
     onUpdateGoals(copyOfSelectedGoals);
+
+    // if we have no goals, open the form up via the
+    // hander provided by the context
+    if (copyOfSelectedGoals.length === 0) {
+      toggleGoalForm(false);
+    }
   };
```

Comment:
check if the newCopyOfSelectedGoals is empty, and if so, reopen the form via the handler provided in GoalFormContext